pkg/platform/runtime/runtime.go

```diff
@@ -66,9 +65,26 @@ func New(target setup.Targeter) (*Runtime, error) {
 	return r, err
 }
 
-func (r *Runtime) Update(msgHandler setup.MessageHandler) error {
+type UpdateMessageHandler interface {
+	HandleUpdateEvents(eventCh <-chan events.BaseEventer)
```

Comment:
This is breaking the promise of the runtime message handler imo. Why do we have callbacks as well as channels? Use one or the other. Preferrably callbacks, leave the channels inside the runtime package.